graphwalker-java/src/main/java/org/graphwalker/java/test/Result.java

```diff
@@ -201,7 +201,7 @@ public void updateResults(Machine machine, Map<Context, MachineException> failur
       results.put("requirementsFailed", requirementsFailedJson);
     }
 
-    if (failures.size() > 0) {
+    if (failures != null && failures.size() > 0) {
```

Comment:
In core we have the Objects helper class that gives you a isNotNullOrEmpty(failures) method, maybe it's better to use that from a code readability point of view